### PR TITLE
Add `.github/settings.yml`

### PR DESCRIPTION
**Pull request by chartgerink**
We recently installed [the GitHub settings application](https://probot.github.io/apps/settings/) to the `epiverse-trace` organization. This PR adds the relevant `.github/settings.yml` file that will then become part of any future repo's built on the template.

This PR **must** be reviewed, as it also has consequences for this repository that need to be resolved first (see below). The settings are my first best guess as to what we want, based on existing settings from the repositories that I investigated. These are only a departure point for our discussion here 💙 

## Questions

- [x] The settings set wiki + discussions to false by default - this repo has content in both. Depending on whether we want to disable the wiki + discussions by default, how do we deal with the discussion and wiki that currently exist here?

### CODE REVIEW

File: .github/settings.yml
Comment (Bisaloo) [inline .github/settings.yml:12]:
```suggestion
  has_issues: true
```

Since all settings are optional, can we leave the ones that don't correspond to an official policy up to the specific maintainers?

In particular, we should not overwrite the private setting, as this can cause serious issues.

File: .github/settings.yml
Comment (Bisaloo) [inline .github/settings.yml:18]: We don't necessarily require approvals on all PR everywhere. Can we set it to `null`? I don't fully understand the docs here.

File: .github/settings.yml
Comment (Bisaloo) [reply .github/settings.yml:12]: My doubt here is: what does "optional" mean? Will it rely on GitHub's defaults, or use this bot defaults?

File: .github/settings.yml
Comment (Bisaloo) [inline .github/settings.yml:1]: It would be good if we can also set the GitHub Pages settings in accordance with our pkgdown workflow. Let's keep an eye on https://github.com/repository-settings/app/issues/609.

File: .github/settings.yml
Comment (Bisaloo) [inline .github/settings.yml:16]: FYI:
We have discussed using `required_linear_history: true` with @bahadzie in the past but didn't come up with a satisfying solution.
The problem is that it will **not** remove the button "update branch" (implied: with a merge commit), but then block you from merging the PR is you clicked this button.

File: .github/settings.yml
Comment (chartgerink) [reply .github/settings.yml:12]: Optional I think will default to GitHub's defaults, but it is not explicitly documented.

Regarding the settings, I am happy to leave it up to the maintainers - it will add importance to actually defining maintainers though 😊

**Review by Bisaloo — APPROVED** [review]

**Pull request merged**
